src/main/java/codeu/controller/ProfileServlet.java

```diff
@@ -20,25 +23,53 @@
  */
 public class ProfileServlet extends HttpServlet {
 
+    private UserStore userStore;
     /**
      * Set up state for handling profile-related requests. This method is only called when
      * running in a server, not when running in a test.
      */
     @Override
     public void init() throws ServletException {
         super.init();
+        setUserStore(UserStore.getInstance());
     }
 
+    void setUserStore(UserStore userStore) {
+      this.userStore = userStore;
+    }
     /**
      * This function fires when a user requests the /users/ URL. It gets the user from the
      * URL and forwards the request to profile.jsp.
      */
     public void doGet(HttpServletRequest request, HttpServletResponse response)
             throws IOException, ServletException {
+
         String username = (String)request.getSession().getAttribute("user");
         request.setAttribute("username", username);
         String requestUrl = request.getRequestURI();
         String user = requestUrl.substring("/users/".length());
+
+        //best friend logic
+        String bf = "no best friend yet";
+        int kwmatching = 0;
+        int kwmatched = 0;
+        List<User> users = userStore.getAllUsers();
+        User userobj = null;
+        userobj = userStore.getUser(username);
```

Comment:
Can be combined into one line with line 57. Also, do you need to check for null/empty username?